src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -0,0 +1,149 @@
+        }
+
+        protected internal override void OpenPin(int pinNumber)
+        {
+            SafeLineHandle pinHandle = Interop.GetChipLineByOffset(_chip, pinNumber);
+            if (pinHandle == null)
+            {
+                throw new IOException($"Pin number {pinNumber} not available for chip: {_chip} error: {Marshal.GetLastWin32Error()}");
+            }
+            _pinNumberToSafeLineHandle.Add(pinNumber, pinHandle);
+        }
+
+        protected internal override PinValue Read(int pinNumber)
+        {
+            if (_pinNumberToSafeLineHandle.TryGetValue(pinNumber, out SafeLineHandle pinHandle))
+            {
+                int value = Interop.GetGpiodLineValue(pinHandle);
+                if (value == -1)
+                {
+                    throw new IOException($"Error while reading value from pin number: {pinNumber}, error: {Marshal.GetLastWin32Error()}");
+                }
+                return value;
+            }
+            throw new IOException($"Error while reading value from pin number: {pinNumber}");
+        }
+
+        protected internal override void RemoveCallbackForPinValueChangedEvent(int pinNumber, PinChangeEventHandler callback)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected internal override void SetPinMode(int pinNumber, PinMode mode)
+        {
+            int success = -1;
+            if (_pinNumberToSafeLineHandle.TryGetValue(pinNumber, out SafeLineHandle pinHandle))
+            {
+                if (mode == PinMode.Input)
+                {
+                    success = Interop.RequestLineInput(pinHandle, pinNumber.ToString());
+                }
+                else
+                {
+                    success = Interop.RequestLineOutput(pinHandle, pinNumber.ToString());
+                }
+            }
+
+            if (success == -1) {
+                throw new IOException($"Error setting pin mode, pin:{pinNumber}, error: {Marshal.GetLastWin32Error()}");
+            } 
+        }
+
+        protected internal override WaitForEventResult WaitForEvent(int pinNumber, PinEventTypes eventType, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected internal override void Write(int pinNumber, PinValue value)
+        {
+            if (_pinNumberToSafeLineHandle.TryGetValue(pinNumber, out SafeLineHandle pinHandle))
+            {
+                Interop.SetGpiodLineValue(pinHandle, (value == PinValue.High) ? 1 : 0);
+            }
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            foreach (SafeLineHandle pinHandle in _pinNumberToSafeLineHandle.Values)
+            {
+                if (pinHandle != null)
+                {
+                    Interop.ReleaseGpiodLine(pinHandle);
+                }
+            }
+
+            if (_chip != null)
+            {
+                _chip.Dispose();
```

Comment:
NIT: Set to null after so that if disposed gets called twice, then this won't start having wierd outcomes.